src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -393,50 +395,77 @@ public void Initialize()
             using (ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
             {
                 if (TryLoadAuthenticationKeyPairs(ctx) == false)
-                    GenerateAuthenticationKeyPairs(ctx);
+                    GenerateAuthenticationSignetureKeys(ctx);                
             }
+            GenerateAuthenticationBoxKeys();
 
             Task.Run(ClusterMaintenanceSetupTask, ServerShutdown);
         }
 
         public byte[] BoxPublicKey, BoxSecretKey, SignPublicKey, SignSecretKey;
 
-        private void GenerateAuthenticationKeyPairs(TransactionOperationContext ctx)
+        private Tuple<DateTime, string> _clusterToken = Tuple.Create<DateTime, string>(DateTime.MinValue, null);
```

Comment:
I didn't used named tuple because i can't replace them atomically with interlocked this is by design